tests/test_backend_loki_field_modifiers.py

```diff
@@ -0,0 +1,113 @@
+from typing import Any, Dict, Tuple
+import pytest
+from sigma.backends.loki import LogQLBackend
+from sigma.collection import SigmaCollection
+from sigma.exceptions import SigmaFeatureNotSupportedByBackendError, SigmaTypeError
+from sigma.modifiers import modifier_mapping
+from sigma.processing.pipeline import ProcessingPipeline
+
+
+@pytest.fixture
+def loki_backend() -> LogQLBackend:
+    # Add a processing pipeline to the backend to test the expand modifier.
+    pipeline = ProcessingPipeline.from_yaml(
+        """
+        name: Test Pipeline
+        priority: 20
+        vars:
+            test: valueA
+        transformations:
+            - type: value_placeholders
+        """
+    )
+    return LogQLBackend(processing_pipeline=pipeline)
+
+
+# Mapping from modifier identifier strings to modifier classes
+modifier_sample_data: Dict[str, Tuple[Any, str]] = {
+    # "modifer": (value, expected_output)
+    "contains": ("valueA", "fieldA=~`(?i).*valueA.*`"),
+    "startswith": ("valueA", "fieldA=~`(?i)valueA.*`"),
+    "endswith": ("valueA", "fieldA=~`(?i).*valueA`"),
+    "exists": ("yes", 'fieldA!=""'),
+    "base64": ("valueA", "fieldA=~`(?i)dmFsdWVB`"),
+    "base64offset": (
+        "valueA",
+        "fieldA=~`(?i)dmFsdWVB` or fieldA=~`(?i)ZhbHVlQ` or fieldA=~`(?i)2YWx1ZU`",
+    ),
+    "wide": ("valueA", "fieldA=~`(?i)v\x00a\x00l\x00u\x00e\x00A\x00`"),
+    "windash": ("-foo", "fieldA=~`(?i)\\-foo` or fieldA=~`(?i)/foo`"),
+    "re": (".*valueA$", "fieldA=~`.*valueA$`"),
+    "i": ("valueA", "valueA"),
```

Comment:
Hmmm, I don't think this is working correctly - I would expect the value to be prefixed with `(?i)`. Don't worry about it here, will reopen #154

Reply:
Correct.